### ISSUE TITLE
Add action to show inline all diagnostics in file.

### ISSUE THREAD

**Issue by Krzysztof-Cieslak** [Microsoft/vscode#19749]
`F8` and `Shift + F8` navigation between diagnostic is really nice but sometimes it would be useful to have nice view of all diagnostics in current file. 

Looks like it would be fairly simple to implement, just one additional editor action in https://github.com/Microsoft/vscode/blob/2a40036dbb34aac4f5aefbe999c21f2eaf76c59b/src/vs/editor/contrib/gotoError/browser/gotoError.ts, I could try implementing it if this feature idea is accepted.

**Comment by bpasero** [comment]
@Krzysztof-Cieslak did you try the Problems view? It shows below the editor all problems.

/cc @sandy081 @alexandrudima

**Comment by Krzysztof-Cieslak** [comment]
Yes, I know about Problems view. But I think that having inlined errors can be useful anyway. :)

![image](https://cloud.githubusercontent.com/assets/5427083/22589108/d9bd8558-ea08-11e6-99f4-a8acda326b80.png)


Changes in code are rather small so it's not adding any complexity etc, it's command so it's opt-in by users, it's providing same UI as existing diagnostics navigation, I don't see any drawbacks
https://github.com/Microsoft/vscode/compare/master...Krzysztof-Cieslak:inline-all-decorations

**Comment by Krzysztof-Cieslak** [comment]
This is pretty much ready, If there is decision that feature proposal is accepted, I'll make PR.

**Comment by sandy081** [comment]
@jrieken FYI

**Comment by jrieken** [comment]
I like but there some technical challenges because of some shortcomings of the current implementation. The problem is that the zone widget with the little arrow on top can only be shown when its target line is rendered. Therefore it always reveals that line to enforce rendering. That's not a problem with how we use it today because goto error reveals the error anyways. However, it's a different story when showing all errors, the last one will always be revealed... 

In short, I'd happily review/guide a PR for this but it is a little bit of work...

**Comment by Gruntfuggly** [comment]
There is an extension in the marketplace called ErrorLens which does this: https://marketplace.visualstudio.com/items?itemName=PhilHindle.errorlens

Might be useful until it's supported natively...

**Comment by usernamehw** [comment]
Forked ErrorLens and made a few tweaks:
1. Update decorations for all visible editors (split/grid) instead of only active one
1. Hot reload of settings
1. Support of gutter icons
1. Set priority of decorations right: Error => Warning => Info => Hint
1. Some other features like (Excluding problems; Update only on save; Update with delay; Set colors for light theme, ...)

https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens


![](https://github.com/usernamehw/vscode-error-lens/raw/master/img/demo.png)

**Comment by DavidGoldman** [comment]
Any update here? It's been about a year

**Comment by Prajwal-Koirala** [comment]
Hello,
Anyone got an update on this?
Really looking forward to this.
- prajwal

**Comment by GrayXu** [comment]
For anyone who needs this feature, [Errors Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens)
